protocol/src/common/protocol.ts

```diff
@@ -4094,6 +4094,14 @@ export interface WorkspaceEditClientCapabilities {
 	 * @since 3.16.0
 	 */
 	changeAnnotationSupport?: ChangeAnnotationsSupportOptions;
+
+	/**
+	 * Whether the client supports snippets as text edits.
+	 *
+	 * @since 3.18.0
+	 * @proposed
+	 */
+	snippetEditSupport?: boolean;
```

Comment:
One additional thing. We need to initialize that capability in the client somewhere here: https://github.com/microsoft/vscode-languageserver-node/blob/ccc901de19515f76521152b0282a90178271e110/client/src/common/client.ts#L1835